addons/calendar/models/calendar_event.py

```diff
@@ -276,6 +276,15 @@ def _compute_stop(self):
             if event.allday:
                 event.stop -= timedelta(seconds=1)
 
+    @api.onchange('start_date', 'stop_date')
+    def _onchange_date(self):
+        for event in self:
+            if event.stop_date and event.start_date:
+                event.write({
+                    'start': fields.Datetime.from_string(event.start_date).replace(hour=8),
+                    'stop': fields.Datetime.from_string(event.stop_date).replace(hour=18),
+                })
```

Comment:
I would put a comment to explain why this is needed as it is quite similar to the inverse method.
Note: the inverse only handle the case of the all day events that may be why there is a traceback :thinking: 

Another possibility would be to update the inverse method but there could be some side effects.

Reply:
I have added the comment explaining why we need that. The inverse is not even called, so we can't do much there.